Helper/Data.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+
+namespace MagentoHackathon\Toolbar\Helper;
+
+use Magento\Framework\App\Helper\Context;
+use Magento\Framework\App\Helper\AbstractHelper;
+use Magento\Framework\App\State;
+use Magento\Store\Model\StoreManagerInterface;
+
+class Data extends AbstractHelper
+{
+    /**
+     * Get the url for this module
+     * @param  string $route
+     * @param  array $params
+     * @return string
+     */
+    public function getUrl($route = '', $params = [])
+    {
+        $route = 'hackathon_toolbar/' . $route;
+
+        return $this->_getUrl($route, $params);
+    }
+
+    /**
+     * Get a Config value
+     *
+     * @param $path
+     * @return mixed
+     */
+    public function getConfigValue($path)
+    {
+        return $this->scopeConfig->getValue($path);
+    }
+
+    /**
+     * Get the current Request
+     *
+     * @return \Magento\Framework\App\RequestInterface
+     */
+    public function getRequest()
+    {
+        return $this->_getRequest();
```

Comment:
I would not do that, but instead use my suggestion, to add the Request as a service dependency to the classes that need them. 

I would not add the Helper class at all but split it up into different services. I think the concept of an "Helper"  is there in Magento 2 to make it simpler to convert Magento 1 code, but it shouldn't be used for new modules. 

In a services oriented architecture like Magento 2 is, each service should do one thing, and do this one thing in a great way. A service should never have more than one responsibility. The reason is that it should be easy to switch out a service if the logic of this service should be changed. This is easily done if the service has one function. In the case that a service has two or more responsibilies you create the problem that you can create the situation that two classes want to override the same service just to change a bit of behavior of this service. In this case you are forced to create yet an other class that merges the two or more changes in a new class.

The next problem with a big helper class is that it makes testing harder, in many cases when you want to create a test you want to create a specific functionality without requiring to much external environment. If you have big helper class (or even extend  from AbstractHelper) you need to know how this class must be overridden making a mock replacement class very hard.

The last problem with creating a helper or a big utility class is that it hides it's dependencies. A class with dependencies does to much in most cases. I would say you should always try to use less than 10 external classes in each class. PHPMD default rules state 13 (https://phpmd.org/rules/design.html#couplingbetweenobjects) which is already quite high to my opinion. I know that Magento 2 does a really crappy job at this, you see @SuppressWarnings(PHPMD.CouplingBetweenObjects) everywhere in the code, and with the creation of "Context"  classes there even hiding a multitude of dependencies.

I would suggest adding a ToolbarStateProvider class that provides the methods that indicate if the toolbar is active and should be displayed. Because this is an class that is used outside the module itself it should have an corresponding interface under the Api namespace (and this Api interface is what you use as type hint).

The request class can be injected in the services that need it.

The configuration retrieval can be something internal and does not need an Api, but it should have a clear name indicating what it does (like ToolbarConfiguration)